nautobot_golden_config/static/run_job.js

```diff
@@ -9,49 +9,63 @@
-          // Normalize to base as much as you can.
-          $('#jobStatus').html("Pending").show();
-          $('#loaderImg').show();
-          $('#jobResults').hide();
-          $('#redirectLink').hide();
-          $('#detailMessages').hide();
-      },
-      success: function(jobData) {
-          $('#jobStatus').html("Started").show();
-          var jobResultUrl = "/extras/job-results/" + jobData.job_result.id + "/";
-          $('#jobResults').html(iconLink(jobResultUrl, "mdi-open-in-new", "Job Details")).show();
-          pollJobStatus(jobData.job_result.url, callBack);
-          if (typeof redirectUrlTemplate !== "undefined") {
-              var redirectUrl = _renderTemplate(redirectUrlTemplate, jobData);
-              $('#redirectLink').html(iconLink(redirectUrl, "mdi-open-in-new", "Info"));
-          }
-      },
-      error: function(e) {
-          $("#loaderImg").hide();
-          console.log("There was an error with your request...");
-          console.log("error: " + JSON.stringify(e));
-          $('#jobStatus').html("Failed").show();
-          $('#detailMessages').show();
-          $('#detailMessages').attr('class', 'alert alert-danger text-center');
-          $('#detailMessages').html("<b>Error: </b>" + e.responseText);
-      }
-  });
+    $.ajax({
+        type: 'POST',
+        url: jobApi,
+        contentType: "application/json",
+        data: JSON.stringify(data),
+        dataType: 'json',
+        headers: {
+            'X-CSRFToken': nautobot_csrf_token
+        },
+        beforeSend: function () {
+            // Normalize to base as much as you can.
+            $('#jobStatus').html("Pending").show();
+            $('#loaderImg').show();
+            $('#jobResults').hide();
+            $('#redirectLink').hide();
+            $('#detailMessages').hide();
+        },
+        success: function (jobData) {
+            if (jobData.job_result) {
+                $('#jobStatus').html("Started").show();
+                var jobResultUrl = "/extras/job-results/" + jobData.job_result.id + "/";
+                $('#jobResults').html(iconLink(jobResultUrl, "mdi-open-in-new", "Job Details")).show();
+                pollJobStatus(jobData.job_result.url, callBack);
+                if (typeof redirectUrlTemplate !== "undefined") {
+                    var redirectUrl = _renderTemplate(redirectUrlTemplate, jobData);
+                    $('#redirectLink').html(iconLink(redirectUrl, "mdi-open-in-new", "Info"));
+                }
+            } else if (jobData.scheduled_job && jobData.scheduled_job.approval_required) {
+                $("#loaderImg").hide();
+                $('#jobStatus').html("Pending Approval").show();
+                $('#detailMessages').show();
+                $('#detailMessages').html(`<p>Job is pending approval, please review <a href="/extras/jobs/scheduled-jobs/approval-queue/${jobData.scheduled_job.id}/">Job Approval</a></p>`);
+            } else {
```

Comment:
```suggestion
            // Job failed, but could be any device that did so, so, not let them check themselves
            } else {
```

Reply:
The message shown seems like otherwise

```js
$('#detailMessages').html("<p>The call to launch the job was successful but unable to determine job status, response has been logged to the browser console.</p>");
```

Reply:
Right now the else should never be able to be hit. If the job launches it takes the first if, if approval required is else if, no know condition would be else. Just to prevent a permanent spinning circle

Reply:
I'm just saying, the message should be in line with the condition.